OpenRA.Mods.Common/EditorBrushes/EditorDefaultBrush.cs

```diff
@@ -51,6 +52,15 @@ public EditorDefaultBrush(EditorViewportControllerWidget editorWidget, WorldRend
 				.ToDictionary(r => r.Info.ResourceType, r => r);
 		}
 
+		long CalculateActorSelectionPriority(EditorActorPreview actor)
+		{
+			var center = new int2(actor.Bounds.X, actor.Bounds.Y);
+			long priority = (center - worldPixel).Length;
+			priority <<= 32;
```

Comment:
You could also make it more explicit that you're not doing anything magic by combining it into `return (long)((center - worldPixel).Length) << 32 + actor.CenterPosition.Z;` instead of successively modifying a temporary var.